packages/chats/src/Lightbox/DuplicateContent.tsx

```diff
@@ -0,0 +1,210 @@
+import { Zoom } from './Zoom';
+import { UIConstant } from '../constants';
+
+export const DuplicateContent = ({
+    fullSizeImage,
+    previewImage,
+    onClose,
+    originalRef,
+}: DuplicateContentProps) => {
+    const theme = useTheme();
+    const insets = useSafeAreaInsets();
+
+    useStatusBar({
+        backgroundColor: ColorVariants.StaticBlack,
+    });
+
+    const [isFullSizeDisplayed, setIsFullSizeDisplayed] = React.useState<boolean>(false);
+
+    /**
+     * This state is used to create transparency during the "swipeToClose" event
+     */
+    const underlayOpacity = useSharedValue<number>(VisibilityState.Opened);
+
+    const {
+        duplicateContentState,
+        onPressClose,
+        onMeasureEnd,
+        onLayout,
+    } = useDuplicateContentState(isFullSizeDisplayed, setIsFullSizeDisplayed);
+
+    useBackHandler(() => {
+        onPressClose();
+        return true;
+    });
+
+    const { pageY, pageX, width, height } = useDimensions(
+        originalRef,
+        duplicateContentState,
+        onMeasureEnd,
+    );
+
+    const onAnimationEnd = useOnAnimationEnd(onClose, setIsFullSizeDisplayed);
+
+    const visibilityState = useVisibilityState(duplicateContentState, onAnimationEnd);
+
+    const { animatedContainerStyle, openedImageScale } = useAnimatedContainerStyle(
+        visibilityState,
+        pageY,
+        pageX,
+        width,
+        height,
+    );
+
+    const contentWidth = useDerivedValue(() => {
+        return width.value * openedImageScale.value;
+    });
+    const contentHeight = useDerivedValue(() => {
+        return height.value * openedImageScale.value;
+    });
+
+    const previewImageStyle = useAnimatedStyle(() => {
+        return {
+            opacity:
+                duplicateContentState.value === DuplicateContentState.Initial ||
+                duplicateContentState.value === DuplicateContentState.Measurement ||
+                visibilityState.value === VisibilityState.Closed
+                    ? 0
+                    : 1,
+        };
+    }, []);
+
+    const underlayStyle = useAnimatedStyle(() => {
+        const resultVisibilityState = underlayOpacity.value * visibilityState.value;
+        return {
+            opacity: interpolate(
+                resultVisibilityState,
+                [VisibilityState.Closed, VisibilityState.Opened],
+                [0, 1],
+            ),
+        };
+    }, []);
+
+    const headerStyle = useAnimatedStyle(() => {
+        const resultVisibilityState = underlayOpacity.value * visibilityState.value;
+        return {
+            opacity: interpolate(
+                resultVisibilityState,
+                [VisibilityState.Closed, VisibilityState.Opened],
+                [0, 1],
+            ),
+        };
+    }, []);
+
+    const styles = useStyles(theme, insets);
+
+    return (
+        <Portal absoluteFill>
+            <Animated.View style={styles.duplicateContainer}>
```

Comment:
Actually I created components with `Content` suffix to avoid updating portals when re-render happens, as with children changes it [calls `setState` internally](https://github.com/tonlabs/UIKit/blob/development/packages/hydrogen/src/Portal.tsx#L156-L164) and avoiding it might impact performance :)

Reply:
Like [there](https://github.com/tonlabs/UIKit/blob/development/packages/navigation/src/Sheets/UISheet/UISheet.tsx#L272-L275)

Reply:
Portal moved from `DuplicateContent` to `Duplicate`.

Reply:
Btw, I didn't notice, did you handle back button on Android there? :)

Reply:
Yes, I used useBackHandler

Reply:
> Yes, I used useBackHandler

Please make sure that you used it in `Content` component :)

Reply:
[I am :)](https://github.com/tonlabs/UIKit/pull/1203/files/f11d0a929ba85ab31169154a06bbd575920d2e4e#diff-2c437319c3416ceba6f9f52c2e1f9474d84beebf462e18bee911b08508b3cc46R62-R65)